### ISSUE TITLE
Odd/unexpected behaviour with `is.na`

### ISSUE THREAD

**Issue by iago-pssjd** [Rdatatable/data.table#6192]
I maybe should expect this, but I didn't:
```r
dt = data.table(x = c(1, NA))
class(dt[, is.na(.SD)])
[1] "matrix" "array"
```
Shouldn't be a data.table?

Thanks!

**Comment by iago-pssjd** [comment]
Well, actually, if it is equivalent to `is.na(dt)` (and there is no a `data.table` method for it), then I must expect that behaviour. There is no other example.

Thanks!

**Comment by ben-schwen** [comment]
There is a `is.na.data.table` so the request would be for `is.na.data.table` to return a `data.table`?

**Comment by MichaelChirico** [comment]
we do the same as data.frame here: is.na(mtcars)

**Comment by iago-pssjd** [comment]
> There is a is.na.data.table so the request would be for is.na.data.table to return a data.table?
@ben-schwen  Well, it was more an expectation (actually, mainly for `dt[, is.na(.SD)]` and distinct to `is.na(dt)`), than a request, but the idea was that.

> we do the same as data.frame here: is.na(mtcars)

@MichaelChirico It's fine to me.

**Comment by iago-pssjd** [comment]
@ben-schwen @MichaelChirico Although being the default same behaviour than for data.frame, it would be great having an option to return a data.table...

**Comment by ben-schwen** [comment]
Could you provide an example where `is.na(x, as.data.table=TRUE)` is advantageous over `as.data.frame(is.na(x))`. Personally, I use `is.na` mainly for indexing so I like that it returns a matrix

**Comment by iago-pssjd** [comment]
Indeed, my solution was to do `dt[, data.table(is.na(.SD))]`, but this implies a double conversion (copy) data.table -> matrix -> data.table. Not an example, but I will speculate  a bit. Wouldn't there be a more direct and efficient way (let's say, something "by reference") so that `is.na(x, as.data.table=TRUE)` there will not make a matrix to get a data.table?

**Comment by MichaelChirico** [comment]
Note that `is.na()` primitive does not accept new arguments (there's no `...`), so I'm not sure it's possible (and probably not recommended) to extend it with such an argument.

Note that the implementation is pretty trivial:

https://github.com/Rdatatable/data.table/blob/6df30c39cc5d1e910e6f4ab84ccdcc693c64315c/R/data.table.R#L2378

So that if you want this, I recommend just doing `setDT(lapply(x, is.na))`.

**Comment by iago-pssjd** [comment]
Good to know. Thanks!
